###Implementation of Linear Regression###

In [1]:
import pandas as pd

In [3]:
df = pd.read_csv("regression1.csv")

In [11]:
df


Math  Physics
0    28       26
1    43       35
2    25       24
3    72       53
4    78       60
5    72       54
6    90       71
7    91       68

In [16]:
df.Physics
df.Math

0    28
1    43
2    25
3    72
4    78
5    72
6    90
7    91
Name: Math, dtype: int64

In [60]:
# X as as independent variable
# y as a depependent varible
X = df.Physics.values.reshape((-1,1))
y = df.Math.values

In [66]:
X

array([[26],
       [35],
       [24],
       [53],
       [60],
       [54],
       [71],
       [68]])

In [65]:
# X is an array
X.shape

(8, 1)

In [62]:
import numpy as np 
from sklearn.linear_model import LinearRegression

In [32]:
model = LinearRegression()
model.fit(X,y) 

LinearRegression()

In [51]:
# coef stand for coefficients = a
a = model.coef_

In [55]:
a = list(a)
a[0]

1.4452930257226195

In [35]:
b = model.intercept_

-8.263696632193032

Equation y = ax+b 
         y = 1.445x - 8.26

ถ้านักเรียนคนที่9 ได้คะแนนphysics 60คะแนน เขาน่าจะได้คะแนนmathเท่าไร

In [72]:
y_pred = model.predict(np.array([[60]]))

In [40]:
#Answer ~ 78
y_pred

array([78.45388491])

In [45]:
from sklearn import metrics

In [46]:
y_pred = model.predict(X)

In [47]:
MSE = metrics.mean_squared_error(y,y_pred)
RMSE = np.sqrt(MSE)

In [48]:
MSE

5.333121188013797

In [49]:
RMSE

2.3093551454927406